localForage and persistCache type collision

Following code produces a TypeScript error: 
```ts
import localForage from 'localforage';
import { persistCache } from 'apollo-cache-persist';
import { InMemoryCache } from 'apollo-cache-inmemory';

const cache = new InMemoryCache();


persistCache({
  cache,
  storage: localForage,
});
```

```

src/shared/apollo.ts:10:14 - error TS2345: Argument of type '{ cache: InMemoryCache; storage: LocalForage; }' is not assignable to parameter of type'ApolloPersistOptions<NormalizedCacheObject>'.
  Types of property 'storage' are incompatible.
    Type 'LocalForage' is not assignable to type 'PersistentStorage<PersistedData<NormalizedCacheObject>>'.
      Types of property 'setItem' are incompatible.
        Type '<T>(key: string, value: T, callback?: ((err: any, value: T) => void) | undefined) => Promise<T>' is not assignable to type '(key: string, data: PersistedData<NormalizedCacheObject>) => void | Promise<void>'.
          Type 'Promise<PersistedData<NormalizedCacheObject>>' is not assignable to type 'void | Promise<void>'.
            Type 'Promise<PersistedData<NormalizedCacheObject>>' is not assignable to type 'Promise<void>'.
              Type 'PersistedData<NormalizedCacheObject>' is not assignable to type 'void'.
                Type 'null' is not assignable to type 'void'.

10 persistCache({
                ~
11   cache,
   ~~~~~~~~
12   storage: localForage,
   ~~~~~~~~~~~~~~~~~~~~~~~
13 });
   ~
```

--- a/src/types/index.ts
+++ b/src/types/index.ts
@@ -12,7 +12,7 @@ export type PersistedData<T> = T | string | null;
 
 export interface PersistentStorage<T> {
   getItem: (key: string) => Promise<T> | T;
-  setItem: (key: string, data: T) => Promise<void> | void;
+  setItem: (key: string, data: T) => Promise<void | T> | void | T;
   removeItem: (key: string) => Promise<void> | void;
 }
 

